plugins/GSdx/GSPng.cpp

```diff
@@ -20,15 +20,59 @@
 
 #include "stdafx.h"
 #include "GSPng.h"
+#include <png.h>
 
 namespace GSPng {
 
+    void SaveFile(const string& file, uint8** image, uint32 width,
+            uint32 height, int channel_bit_depth, int type, bool rb_swapped = false)
+    {
+        FILE *fp = fopen(file.c_str(), "wb");
+        if (fp == nullptr)
+            return;
+
+        png_structp png_ptr = png_create_write_struct(PNG_LIBPNG_VER_STRING,
+                nullptr, nullptr, nullptr);
+        if (png_ptr == nullptr) {
+            fclose(fp);
+            return;
+        }
+
+        png_infop info_ptr = png_create_info_struct(png_ptr);
+        if (info_ptr == nullptr) {
+            png_destroy_write_struct(&png_ptr, nullptr);
+            fclose(fp);
+            return;
+        }
+
+        if (setjmp(png_jmpbuf(png_ptr))) {
+            png_destroy_write_struct(&png_ptr, nullptr);
+            fclose(fp);
+            return;
+        }
+
```

Comment:
The brace placement is similar to the K&R/Linux kernel coding style - function/scope braces on a separate line, every other brace on the same line (switch/for/if).

I prefer braces on the same line simply because I can see more lines. IMO it doesn't make things harder to spot - I rely a lot more on indentation for that.